Sources/Commands/SwiftTool.swift

```diff
@@ -696,6 +746,124 @@ extension SwiftTool: BuildPlanDelegate {
+    }
+
+    /// Called when a command is preparing to start.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandPreparing(_ command: Command) {
+    }
+
+    /// Called when a command has been started.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandStarted(_ command: Command) {
+        guard command.shouldShowStatus else { return }
+        print(command.description)
+    }
+    
+    /// Called to allow the delegate to skip commands without cancelling their
+    /// dependents. See llbuild's should_command_start.
+    public func shouldCommandStart(_ command: Command) -> Bool {
+        return true
+    }
+
+    /// Called when a command has been finished.
+    public func commandFinished(_ command: Command, result: CommandResult) {
+    }
+
+    /// Called to report an error during the execution of a command.
+    public func commandHadError(_ command: Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandErrorDiagnostic(message: message))
+    }
+
+    /// Called to report a note during the execution of a command.
+    public func commandHadNote(_ command: Command, message: String) {
+    }
+
+    /// Called to report a warning during the execution of a command.
+    public func commandHadWarning(_ command: Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandWarningDiagnostic(message: message))
+    }
+
+    /// Called when a command's job has started executing an external process.
+    ///
+    /// The system guarantees that any commandProcessStarted() call will be paired
+    /// with exactly one \see commandProcessFinished() call.
+    ///
+    /// - parameter process: A unique handle used in subsequent delegate calls
+    /// to identify the process. This handle should only be used to associate
+    /// different status calls relating to the same process. It is only
+    /// guaranteed to be unique from when it has been provided here to when it
+    /// has been provided to the \see commandProcessFinished() call.
+    public func commandProcessStarted(_ command: Command, process: ProcessHandle) {
+    }
+
+    /// Called to report an error in the management of a command process.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter message: The error message.
+    public func commandProcessHadError(_ command: Command, process: ProcessHandle, message: String) {
+    }
+
+    /// Called to report a command processes' (merged) standard output and error.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter data: The process output.
+    public func commandProcessHadOutput(_ command: Command, process: ProcessHandle, data: [UInt8]) {
+        print(String(bytes: data, encoding: .utf8)!)
```

Comment:
This shouldn't add a newline (via the implicit print behavior), the data reported here is the raw data.

Reply:
I think you can  use `stdoutStream` here.

Reply:
@aciidb0mb3r any particular reason vs `print(String(bytes: data, encoding: .utf8)!, terminator: "")`?

Reply:
As Daniel said, the reported bytes here will be raw and might not be valid utf8. `stdoutStream` is an `OutputByteStream` and can take bytes directly.

Reply:
We can even move the stream to init and then unit test the delegate!